cypress/integration/chi-js/dropdown.spec.js

```diff
@@ -0,0 +1,213 @@
+describe('Dropdown', function() {
+    before(() => {
+      cy.visit('http://localhost:8000/tests/js/dropdown.html#');
+    });
+
+    describe('Open and closing functionality test for dropdown base', function() {
+        it('Dropdown menu should open when dropdown trigger is clicked', () => {  
+            cy.get('.chi-dropdown')
+            .find('button#dropdown-1.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .find('+.chi-dropdown__menu')
+            .should('be.visible')
+            .should('have.class', '-active')
+        });
+    
+        it('Dropdown menu should close when dropdown trigger is clicked twice', () => {  
+            cy.get('.chi-dropdown')
+            .find('button#dropdown-1.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .find('+.chi-dropdown__menu')
+            .should('be.not.visible')
+        });
+    })
+
+    describe('Open and closing functionality test for dropdown target', function() {
+        it('Dropdown menu should open when dropdown trigger is clicked', () => {  
+            cy.get('button#dropdown-2.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+            .should('be.visible')
+            .should('have.class', '-active')
+            
+        });
+
+        it('Dropdown menu should close when dropdown trigger is clicked', () => {  
+            cy.get('button#dropdown-2.chi-button.chi-dropdown__trigger')
+            .click()
+            .wait(550)
+        });
+    })
+
+    describe('Functionality test for dropdown on hover', function () {
+          it('Dropdown menu should open when hovered over ', () => {
+            cy.get('.chi-dropdown.chi-dropdown__hover')
+            .find('button.chi-dropdown__trigger')
+            .find('+.chi-dropdown__menu')
+            .invoke('show')
+            .wait(550)
+            .should('be.visible')
+            // .should('have.class', '-active')
+          })
+
+          it('Dropdown menu should close when dropdown trigger is clicked', () => {  
+            cy.get('.chi-dropdown.chi-dropdown__hover')
+            .find('button.chi-dropdown__trigger')
+            .find('+.chi-dropdown__menu')
+            .invoke('hide')
+            .wait(550)
+            .should('be.not.visible')
+          });
+    })
+
+    describe('Open and closing functionality test for animated dropdown', function() {
```

Comment:
Open and close behavior of animated Dropdowns is the same as the normal ones.
The difference is that ::after selector of the trigger button that renders the chevron icon get CSS `transform: rotate(180deg);` Let's try checking its presence after the button is clicked. If it's not possible, let's remove the test case.

Reply:
Did you try checking the presence of CSS `transform: rotate(180deg);` when -animated Dropdown trigger is clicked? The current implementation checks Click + Open / Close behavior but not the actual animation of icon rotation.
I put together this example of getting rotation value of an :after pseudo element 180deg rotation:
https://codepen.io/janibolkvadze/pen/OJXbdxz